Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+import numpy as np
+import math
+
+
+class PolDiffILLReduction(PythonAlgorithm):
+
+    _mode = 'Monochromatic'
+    _method_data_structure = None # measurement method determined from the data
+    _user_method = None
+    _instrument = None
+    _experimentProperties = None
+
+    _DEG_2_RAD =  np.pi / 180.0
+
+    @staticmethod
+    def _max_value_per_detector(ws):
+        max_values = np.zeros(mtd[ws].getItem(0).getNumberOfHistograms())
+        for entry in mtd[ws]:
+            for spectrum_no in range(entry.getNumberOfHistograms()):
+                dataY = entry.readY(spectrum_no)
+                if dataY > max_values[spectrum_no]:
+                    max_values = dataY
+        return max_values
+
+    def category(self):
+        return 'ILL\\Diffraction'
+
+    def summary(self):
+        return 'Performs polarized diffraction data reduction at the ILL.'
+
+    def seeAlso(self):
+        return ['PolDIffILLAutoProcess']
+
+    def name(self):
+        return 'PolDiffILLReduction'
+
+    def validateInputs(self):
+        issues = dict()
+        process = self.getPropertyValue('ProcessAs')
+        if process == 'Transmission' and self.getProperty('BeamInputWorkspace').isDefault:
+            issues['BeamInputWorkspace'] = 'Beam input workspace is mandatory for transmission calculation.'
+            issues['CadmiumTransmissionInputWorkspace'] = 'Cadmium transmission input workspace is mandatory for transmission calculation.'
+
+        if ( (not self.getProperty('AbsorberInputWorkspace').isDefault and self.getProperty('ContainerInputWorkspace').isDefault)
+             or (not self.getProperty('AbsorberInputWorkspace').isDefault and self.getProperty('ContainerInputWorkspace').isDefault) ):
+            issues['AbsorberInputWorkspace'] = 'Both Container and Absorber input workspaces are mandatory for background subtraction.'
+            issues['ContainerInputWorkspace'] = 'Both Container and Absorber input workspaces are mandatory for background subtraction.'
+
+        if process == 'Quartz' and self.getProperty('TransmissionInputWorkspace').isDefault:
+            issues['TransmissionInputWorkspace'] = 'Quartz transmission is mandatory for polarization correction calculation.'
+
+        if process == 'Vanadium' and self.getProperty('TransmissionInputWorkspace').isDefault :
+            issues['TransmissionInputWorkspace'] = 'Vanadium transmission is mandatory for vanadium data reduction.'
+
+        if process == 'Sample' or process == 'Vanadium':
+            if self.getProperty('ExperimentPropertiesDictionary').isDefault:
+                issues['ExperimentPropertiesDictionary'] = 'Sample parameters need to be defined.'
+
+            sampleGeometry = self.getProperty('ExperimentPropertiesDictionary').value
+            geometry_type = self.getPropertyValue('SampleGeometry')
+            required_keys = ['mass', 'density', 'number_density', 'beam_height', 'beam_width', 'formula_units', 'container_density',
```

Comment:
You are making up a new set of keys, I think we should stay consistent with the other algorithms and have these in camel case. Look at PaalmanPingsMonteCarloAbsorption for example.